github/client_repository_file.go

```diff
@@ -54,6 +60,22 @@ func (c *FileClient) Get(ctx context.Context, path, branch string) ([]*gitprovid
 
 	for _, file := range directoryContent {
 		filePath := file.Path
+		if *file.Type == "dir" {
+			if recursive == true {
+				if !strings.HasSuffix(path, "/") {
+					path = path + "/"
+				}
+				subdirectoryPath := fmt.Sprintf("%v%v/", path, *file.Name)
+				// recursive call for child directories to get their content
+				childFiles, err := c.Get(ctx, subdirectoryPath, branch, optFns...)
```

Comment:
This is a pretty expensive operation (each file requested is a separate API call).

You are not providing any way to limit it, and there's no way to know up front how many files will be fetched?

Reply:
Using the [repository content](https://docs.github.com/en/rest/repos/contents#get-repository-content) get in github API there is no option to know sub-directory contents, but I believe a call can be added to fetch the [tree](https://docs.github.com/en/rest/git/trees) to know how many files will be fetched. 
But then what decides the limitations or the max number of files/sub-directories to be fetched? will an extra parameter of the max tree length be a good idea?